src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java

```diff
@@ -609,6 +610,7 @@ public void verifyBookingAccess(final Long bookingId, final String... rolesAllow
         if (agencyIds.isEmpty()) {
             if (authenticationFacade.isClientOnly()) {
                 logClientUnauthorisedAccess(bookingId, rolesAllowed);
+                throw new AccessDeniedException(format("Client not authorised to access booking with id %d.", bookingId));
```

Comment:
I originally created a new exception for this, but I think this is fine (especially as I have left in logging/customEvent)